# HG changeset patch
# User Tetsuya Takatsuru <takatsuru.tetsuya@nttcom.co.jp>
# Date 1315359482 -32400
# Branch trunk
# Node ID 9ff7f2c2a4cd0882ee77ca3aa2d32307cbbf53bb
# Parent  91e893dec97660f5b48bc7cb58a5a967b0af9021
port duplicates at Host header when capturing by proxy.

diff -r 91e893dec976 -r 9ff7f2c2a4cd src/protocol/http/org/apache/jmeter/protocol/http/sampler/HTTPHC4Impl.java
--- a/src/protocol/http/org/apache/jmeter/protocol/http/sampler/HTTPHC4Impl.java	木  8月 18 15:40:00 2011 +0900
+++ b/src/protocol/http/org/apache/jmeter/protocol/http/sampler/HTTPHC4Impl.java	水  9月 07 10:38:02 2011 +0900
@@ -630,6 +630,7 @@
                     if (! HEADER_CONTENT_LENGTH.equalsIgnoreCase(n)){
                         String v = header.getValue();
                         if (HEADER_HOST.equalsIgnoreCase(n)) {
+                            String host = v.split(":", 2)[0];
                             // TODO is it a bug that HC 4.x does not add the correct port to the generated Host header?
                             int port = url.getPort();
                             if (port != -1) {
@@ -637,7 +638,7 @@
                                     port = -1; // no need to specify the port if it is the default
                                 }
                             }
-                            request.getParams().setParameter(ClientPNames.VIRTUAL_HOST, new HttpHost(v, port));
+                            request.getParams().setParameter(ClientPNames.VIRTUAL_HOST, new HttpHost(host, port));
                         } else {
                             request.addHeader(n, v);
                         }
